lib/IDE/CodeCompletion.cpp

```diff
@@ -4703,6 +4704,21 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
     }
   }
 
+  void getTypeAttributeKeywordCompletions() {
+    auto addTypeAttr = [&](StringRef Name) {
+      CodeCompletionResultBuilder Builder(
+          Sink,
+          CodeCompletionResult::ResultKind::Keyword,
+          SemanticContextKind::None, expectedTypeContext);
+      Builder.addAttributeKeyword(Name, "Type Attribute");
+    };
+    addTypeAttr("autoclosure");
+    addTypeAttr("convention(swift)");
+    addTypeAttr("convention(block)");
+    addTypeAttr("convention(c)");
```

Comment:
Missing:
```swift
@convention(thin) // no context object, e.g. a global function
```

Technically, there is also an optional `, cType: ` argument on the convention, but I've never seen one used in practice, so maybe it's okay to ignore it here.

Reply:
Oh, I wasn't aware of that convention. Is it officially supported? It's not listed in https://docs.swift.org/swift-book/ReferenceManual/Attributes.html So I hesitate to suggest it.

Reply:
Should be supported AFAIK.  It's in `FunctionTypeRepresentation` and the name isn't underscored, so to me that means supported.  The only conventions I wouldn't expect us to support are the SIL-specific ones, which have "method", "witness_method", and they use the name "thick" instead of "swift".